// @vtex/intl-equalizer/dist/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MESSAGES = exports.ERRORS = void 0;

var _chalk = _interopRequireDefault(require("chalk"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var ERRORS = {
  ERROR_NO_LOCALE_FILES: 'ERROR_NO_LOCALE_FILES',
  ERROR_NO_LOCALE_FOLDER: 'ERROR_NO_LOCALE_FOLDER',
  ERROR_NO_KEYS_LOCALE: 'ERROR_NO_KEYS_LOCALE'
};
exports.ERRORS = ERRORS;
var MESSAGES = {
  COULD_NOT_WRITE_JSON: function COULD_NOT_WRITE_JSON(locale) {
    return _chalk["default"].red("\u274C Error: Could not write keys to ".concat(locale, ".json."));
  },
  NO_LOCALE_FILES: function NO_LOCALE_FILES(pathName) {
    return _chalk["default"].red("\u274C Error: There are no locale files in \"".concat(pathName, ". \n Only base locales are considered. (i.e. LOCALE.json)"));
  },
  NO_LOCALE_FOLDER: function NO_LOCALE_FOLDER(pathName) {
    return _chalk["default"].red("\u274C Error: There are no locale files folder in \"".concat(pathName, "."));
  },
  ERROR_NO_KEYS_LOCALE: function ERROR_NO_KEYS_LOCALE(locale) {
    return _chalk["default"].red("\u274C Error: Locale ".concat(locale, " is empty. Add some keys or delete it."));
  },
  ERROR_COULD_NOT_FIX: function ERROR_COULD_NOT_FIX(_ref) {
    var referenceLocale = _ref.referenceLocale,
        locale = _ref.locale,
        referenceLocaleLength = _ref.referenceLocaleLength,
        localeLength = _ref.localeLength;
    return _chalk["default"].red("\u274C Error: Couldn't fix locale ".concat(locale.toUpperCase(), " due to differences on locale size. \n ").concat(referenceLocale.toUpperCase(), " has ").concat(referenceLocaleLength, " keys. Although, ").concat(locale.toUpperCase(), " has ").concat(localeLength, " keys."));
  },
  WRITING_LOCALE: function WRITING_LOCALE(locale) {
    return _chalk["default"].yellow("\u270F\uFE0F  Writing sorted keys to ".concat(locale, ".json"));
  },
  SUCCESS: _chalk["default"].green('✨ All keys are perfectly localized!'),
  SUCCESS_FIX: '✨ Finished fixing keys!'
};
exports.MESSAGES = MESSAGES;
