// @vtex/intl-equalizer/dist/missingKeysTable.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = createTable;

var _cliTable = _interopRequireDefault(require("cli-table3"));

var _chalk = _interopRequireDefault(require("chalk"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function createTable(result, referenceLocale) {
  var table = new _cliTable["default"]({
    chars: {
      top: '═',
      'top-mid': '╤',
      'top-left': '╔',
      'top-right': '╗',
      bottom: '═',
      'bottom-mid': '╧',
      'bottom-left': '╚',
      'bottom-right': '╝',
      left: '║',
      'left-mid': '╟',
      mid: '─',
      'mid-mid': '┼',
      right: '║',
      'right-mid': '╢',
      middle: '│'
    },
    head: [{
      colSpan: 3,
      hAlign: 'center',
      content: _chalk["default"].yellow("MISSING KEYS")
    }]
  });
  table.push([{
    colSpan: 3,
    hAlign: 'center',
    content: _chalk["default"].yellow("REFERENCE LOCALE: ".concat(referenceLocale.toUpperCase()))
  }]);
  table.push([{
    hAlign: 'center',
    content: _chalk["default"].yellow('LOCALE')
  }, {
    hAlign: 'center',
    content: _chalk["default"].yellow('PATH')
  }, {
    hAlign: 'center',
    content: _chalk["default"].yellow('MISSING KEYS')
  }]);
  Object.keys(result).forEach(function (countryName) {
    if (countryName === referenceLocale) return;
    var country = countryName.toUpperCase();
    var filepath = result[countryName].filepath;
    var missingKeys = result[countryName].missingKeys.reduce(function (acc, curr) {
      return "".concat(acc + curr.key, "\n");
    }, '');
    table.push([country, filepath, missingKeys]);
  });
  console.log(table.toString());
}
