// @vtex/intl-equalizer/dist/wrongOrderKeysTable.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = createTable;

var _cliTable = _interopRequireDefault(require("cli-table3"));

var _chalk = _interopRequireDefault(require("chalk"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function createTable(result, referenceLocale) {
  var table = new _cliTable["default"]({
    chars: {
      top: '═',
      'top-mid': '╤',
      'top-left': '╔',
      'top-right': '╗',
      bottom: '═',
      'bottom-mid': '╧',
      'bottom-left': '╚',
      'bottom-right': '╝',
      left: '║',
      'left-mid': '╟',
      mid: '─',
      'mid-mid': '┼',
      right: '║',
      'right-mid': '╢',
      middle: '│'
    },
    head: [{
      colSpan: 4,
      hAlign: 'center',
      content: _chalk["default"].yellow("KEYS WITH DIFFERENT ORDER \n Run 'intl-equalizer --fix' to fix the order of the keys.")
    }]
  });
  table.push([{
    colSpan: 4,
    hAlign: 'center',
    content: _chalk["default"].yellow("REFERENCE LOCALE: ".concat(referenceLocale.toUpperCase()))
  }]);
  table.push([{
    hAlign: 'center',
    content: _chalk["default"].yellow('LOCALE')
  }, {
    hAlign: 'center',
    content: _chalk["default"].yellow('KEY')
  }, {
    hAlign: 'center',
    content: _chalk["default"].yellow('CURRENT LINE')
  }, {
    hAlign: 'center',
    content: _chalk["default"].yellow('CORRECT LINE')
  }]);
  Object.keys(result).forEach(function (countryName) {
    if (countryName === referenceLocale) return;
    var country = countryName.toUpperCase();
    result[countryName].wrongOrderKeys.forEach(function (currentKey) {
      table.push([country, currentKey.key, currentKey.wrongLine, currentKey.correctLine]);
    });
  });
  console.log(table.toString());
}
